lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -253,6 +253,52 @@ sub update_status {
     $self->render(json => $ret);
 }
 
+sub update {
+    my ($self) = @_;
+
+    my $resultset = $self->app->schema->resultset('Jobs');
+    my $job       = $resultset->find(int($self->stash('jobid')))
+      or return $self->render(json => {error => 'Job does not exist'}, status => 404);
+    my $json = $self->req->json
+      or return $self->render(json => {error => 'No updates provided (must be provided as JSON)'}, status => 400);
+    my $settings = delete $json->{settings};
+
+    # validate specified columns
+    my %allowed_cols = map { $_ => 1 } qw(group_id priority retry_avbl);
+    while (my ($key, $value) = each %$json) {
+        return $self->render(json => {error => 'Column ' . $key . ' can not be set'}, status => 400)
+          unless $allowed_cols{$key};
+    }
+
+    # validate specified group
+    my $group_id = $json->{group_id};
+    if (defined($group_id) && !$self->db->resultset('JobGroups')->find(int($group_id))) {
+        return $self->render(json => {error => 'Group does not exist'}, status => 404);
+    }
+
+    # some settings are stored directly in job table and hence must be updated there
+    my %setting_cols = map { $_ => 1 } qw(TEST DISTRI VERSION FLAVOR ARCH BUILD MACHINE);
+    if ($settings) {
+        for my $setting_col (keys %setting_cols) {
```

Comment:
In the first place I thought it would be required to check whether the setting_cols array contains a string. Since this is not required after all, converting it in a hash is not necessary of course.

> Perhaps you want to use shuffle instead?

No, what I actually want is array_or_set.contains() method and not use hashes where it doesn't make sense.